### PR TITLE
Move platform modules into `sys::pal`

### PR DESCRIPTION
**Pull request by joboet**
This is the initial step of #117276. `sys` just re-exports everything from the current `sys` for now, I'll move the implementations for the individual features one-by-one after this PR merges.

### CODE REVIEW

**Comment by LunarLambda** [comment]
What does `pal` stand for? Platform Abstraction Layer?

**Comment by joboet** [comment]
I've completely rewritten this PR, separating moves and changes, and kept all the comment-editing out, as those will need to be adjusted anyway in the PRs to come.

**Comment by joboet** [comment]
All rebased!

**Comment by ChrisDenton** [comment]
Ok, I skimmed through pre-rebase and it's looking good so I'll do a full review asap.

One thing I note is you'll also need to update the path here (it does code generation): https://github.com/rust-lang/rust/blob/062e7c6a951c1e4f33c0a6f6761755949cde15ec/src/tools/generate-windows-sys/src/main.rs#L19

It would be worth grepping for `src/sys` and seeing if anything else interesting crops up.

**Comment by joboet** [comment]
I feared that there would be much more, but it seems like there just aren't that many comments referencing `sys` by path, so I'll just update them in this PR.

**Comment by ChrisDenton** [comment]
Looks great, thanks! This has a high chance of breaking again so I'll boost its priority and see if we can get merged before that happens

@bors r+ rollup=never p=1

File: library/core/tests/num/int_log.rs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 //! This tests the `Integer::{ilog,log2,log10}` methods. These tests are in a
 //! separate file because there's both a large number of them, and not all tests
 //! can be run on Android. This is because in Android `ilog2` uses an imprecise
-//! approximation:https://github.com/rust-lang/rust/blob/4825e12fc9c79954aa0fe18f5521efa6c19c7539/src/libstd/sys/unix/android.rs#L27-L53
+//! approximation:https://github.com/rust-lang/rust/blob/4825e12fc9c79954aa0fe18f5521efa6c19c7539/src/libstd/sys/pal/unix/android.rs#L27-L53
```
Comment (klensy) [inline library/core/tests/num/int_log.rs:4]: Dead url

Comment (joboet) [reply library/core/tests/num/int_log.rs]: Thank you! I've restored it now.

@rustbot ready

**Pull request merged**